sageintacctsdk/apis/api_base.py

```diff
@@ -468,35 +466,29 @@ def get_all_generator(self, field: str = None, value: str = None, fields: list =
                     }
                 }
 
-            if field and value:
-                data['query']['filter'] = {
-                    'equalto': {
-                        'field': field,
-                        'value': value
-                    }
+            field_value_filter = (
+                {"equalto": {"field": field, "value": value}} if field and value else {}
+            )
+            updated_at_filter = (
+                {
+                    "greaterthanorequalto": (
+                        {"field": "WHENMODIFIED", "value": updated_at}
+                    )
                 }
-
-            field_filter = {}
-            if updated_at and data['query']['filter']:
-                field_filter = data['query']['filter']
-                updated_at_filter = {
-                    'greaterthanorequalto': {
-                        'field': 'WHENMODIFIED',
-                        'value': updated_at
-                    }
-                }
-
-                data['query']['filter'] = {
-                    'and': {}
+                if updated_at
+                else {}
+            )
+            # if we have updated_at_filter and field_value_filter we need to 'and' them
+            if updated_at_filter and field_value_filter:
+                data["query"]["filter"] = {
+                    "and": {**field_value_filter, **updated_at_filter}
                 }
-                data['query']['filter']['and'].update(field_filter)
-                data['query']['filter']['and'].update(updated_at_filter)
-
-            if not data['query']['filter']:
-                del data['query']['filter']
-
-            if not data['query']['orderby']:
-                del data['query']['orderby']
```

Comment:
these blocks are required, else we might get error from APIs

```
{'message': 'Some of the parameters are wrong', 'response': {'error': [{'errorno': 'XL03000003', 'description': None, 'description2': "XML Parse schema error: Error 1871: Element 'orderby': Missing child element(s). Expected is ( order ).. Line: 1, column: 0. [Support ID: XRTTqWEB033~ZnpxnP5j0Br4ja4-a0TLJwAAABE]", 'correction': None}]}} {'error': [{'errorno': 'XL03000003', 'description': None, 'description2': "XML Parse schema error: Error 1871: Element 'orderby': Missing child element(s). Expected is ( order ).. Line: 1, column: 0. [Support ID: XYZ]", 'correction': None}]}

Reply:
See here: https://github.com/fylein/sageintacct-sdk-py/pull/77/files#diff-c3a501ef4ad7ed46ca1eba1244d2d8797e11c44c0978a4d1ee2a8aadc37ca6ccL456-L459, we only set them if we need to, so a delete shouldn't be required.

Reply:
@jtruty we shouldn't send empty keys, the above error came up when I tried to send a request without order by value

Reply:
Got it - updated!